### ISSUE TITLE
fasd interfering with scp on cygwin

### ISSUE THREAD

**Issue by sethfuller** [Bash-it/bash-it#706]
I installed bash-it and I selected fasd as a plugin, but hadn't installed it. I started to get the message:

You must install fasd before you can use this plugin

I then installed fasd, but still got the message. I disabled fasd and still got the message. I uninstalled bash-it and still got the message. I then reinstalled cygwin in another directory and I still get the message. I checked and the new installation has no references to the old installation of cygwin.

When I run scp -v I get this:

debug1: channel 0: new [client-session]
debug1: Requesting no-more-sessions@openssh.com
debug1: Entering interactive session.
debug1: pledge: network
debug1: Sending command: scp -v -t ~/logs
You must install fasd before you can use this plugin
debug1: client_input_channel_req: channel 0 rtype exit-status reply 0
debug1: channel 0: free: client-session, nchannels 1
Transferred: sent 3104, received 3264 bytes, in 1.2 seconds
Bytes per second: sent 2626.1, received 2761.4
debug1: Exit status 0

The command is something like:

scp -v /cygdrive/c/Users/sfx/sfx.out pitx.cx.com:~/logs

This has worked for a year and a half with no problems. I haven't deleted the old /cygwin64 directory, because there are probably things under the directory tree I will want to copy.

**Comment by sethfuller** [comment]
BTW I can ssh to the same server without problem. It is something in the copying that is being interfered with. I'm not sure what the bytes sent and received refer to, but the file is 5926 bytes and is definitely not being transferred. Another time I tried this the bytes were slightly different.

**Comment by nwinkler** [comment]
Thanks for reporting this! Before we look into solving this, let's clarify your setup:

* Where do you have Bash-it installed, on the client, on the server, or on both?
* Same for Cygwin: on the client, on the server, or on both?

The message you're seeing is generated here: https://github.com/Bash-it/bash-it/blob/master/plugins%2Favailable%2Ffasd.plugin.bash#L7

Taking a look at the `scp` output, I suspect that this message is generated on the server in your case. If you have installed Bash-it on the server, then please do the following.

* `ssh` to the server
* Run `bash-it disable plugin fasd`
* Close the `ssh` session
* Try the `scp` command again

I have seen similar instances of this, where output generated by a plugin is interfering with `scp`. The difference between `scp` and `ssh` seems to be the type of session that is opened, and the way server output is handled by the two commands.

Let me know if that helps in your case.

**Comment by sethfuller** [comment]
I am using CentOS 6.6 on my server. Installing fasd on my Linux box seems to have helped. I don't know if I also had to disable nvm (which wasn't installed on my Linux box since scp complained about that too. I didn't notice the file I was trying to copy on my server until I disabled the nvm plugin as well, but I may have missed it).

I am now having a different problem, which I could file a separate report for. I have a script that clones a git repository via ssh and every time it calls ssh it says the script stops and I have to type fg, then enter my ssh password. After every command it says preexec_invoke_exec. I found this in .bash-it/lib/preexec.bash. It looks like this is interfering with the execution of the bash shell function calling ssh. This is on CentOS.
